ydb/core/tx/schemeshard/schemeshard__operation_replace_external_table.cpp

```diff
@@ -0,0 +1,421 @@
+    }
+
+    static bool IsDataSourceValid(const THolder<TProposeResponse>& result,
+                                  const TExternalDataSourceInfo::TPtr& externalDataSource) {
+        if (!externalDataSource) {
+            result->SetError(NKikimrScheme::StatusSchemeError, "Data source doesn't exist");
+            return false;
+        }
+        return true;
+    }
+
+    static bool IsExternalTableDescriptionValid(
+        const THolder<TProposeResponse>& result,
+        const TString& sourceType,
+        const NKikimrSchemeOp::TExternalTableDescription& desc) {
+        if (TString errorMessage; !NExternalTable::Validate(sourceType, desc, errorMessage)) {
+            result->SetError(NKikimrScheme::StatusSchemeError, errorMessage);
+            return false;
+        }
+
+        return true;
+    }
+
+    static void AddPathInSchemeShard(const THolder<TProposeResponse>& result,
+                                     TPath& dstPath) {
+        result->SetPathId(dstPath.Base()->PathId.LocalPathId);
+    }
+
+    TPathElement::TPtr CreateExternalTablePathElement(const TPath& dstPath) const {
+        TPathElement::TPtr externalTable = dstPath.Base();
+        externalTable->PathState = TPathElement::EPathState::EPathStateAlter;
+        externalTable->LastTxId  = OperationId.GetTxId();
+
+        return externalTable;
+    }
+
+    void CreateTransaction(const TOperationContext& context,
+                           const TPathId& externalTablePathId,
+                           const TPathId& externalDataSourcePathId) const {
+        TTxState& txState = context.SS->CreateTx(OperationId,
+                                                 TTxState::TxReplaceExternalTable,
+                                                 externalTablePathId,
+                                                 externalDataSourcePathId);
+        txState.Shards.clear();
+    }
+
+    void RegisterParentPathDependencies(const TOperationContext& context,
+                                        const TPath& parentPath) const {
+        if (parentPath.Base()->HasActiveChanges()) {
+            const auto parentTxId = parentPath.Base()->PlannedToCreate()
+                                   ? parentPath.Base()->CreateTxId
+                                   : parentPath.Base()->LastTxId;
+            context.OnComplete.Dependence(parentTxId, OperationId.GetTxId());
+        }
+    }
+
+    void AdvanceTransactionStateToPropose(const TOperationContext& context,
+                                          NIceDb::TNiceDb& db) const {
+        context.SS->ChangeTxState(db, OperationId, TTxState::Propose);
+        context.OnComplete.ActivateTx(OperationId);
+    }
+
+    static void LinkExternalDataSourceWithExternalTable(
+        const TExternalDataSourceInfo::TPtr& externalDataSource,
+        const TPathElement::TPtr& externalTable,
+        const TPath& dstPath,
+        const TExternalDataSourceInfo::TPtr& oldDataSource,
+        bool IsSameDataSource) {
```

Comment:
`IsSameDataSource` -> `isSameDataSource` - it is parameter

Reply:
Fixed